prov/gni/src/gnix_cq.c

```diff
@@ -612,8 +617,42 @@ DIRECT_FN STATIC ssize_t gnix_cq_readerr(struct fid_cq *cq,
 	}
 
 	event = container_of(entry, struct gnix_cq_entry, item);
-
-	memcpy(buf, event->the_entry, sizeof(struct fi_cq_err_entry));
+	gnix_cq_err = event->the_entry;
+	fabric = container_of(&cq_priv->domain->fabric->fab_fid,
+			      struct util_fabric, fabric_fid);
```

Comment:
This is completely broken.  Why would we be filling in the struct util_provider?  I've opened an upstream issue asking if we should put the API version in the base struct.